src/main/java/uk/gov/hmcts/ccd/domain/service/getcase/AuthorisedGetCaseOperation.java

```diff
@@ -31,43 +34,43 @@
     private final CaseDefinitionRepository caseDefinitionRepository;
     private final AccessControlService accessControlService;
     private final UserRepository userRepository;
+    private final CaseRoleRepository caseRoleRepository;
 
 
     public AuthorisedGetCaseOperation(@Qualifier("classified") final GetCaseOperation getCaseOperation,
                                       @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) final CaseDefinitionRepository caseDefinitionRepository,
                                       final AccessControlService accessControlService,
-                                      @Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository) {
+                                      @Qualifier(CachedUserRepository.QUALIFIER) final UserRepository userRepository,
+                                      @Qualifier(CachedCaseRoleRepository.QUALIFIER) final CaseRoleRepository caseRoleRepository) {
         this.getCaseOperation = getCaseOperation;
         this.caseDefinitionRepository = caseDefinitionRepository;
         this.accessControlService = accessControlService;
         this.userRepository = userRepository;
+        this.caseRoleRepository = caseRoleRepository;
     }
 
     @Override
     public Optional<CaseDetails> execute(String jurisdictionId, String caseTypeId, String caseReference) {
 
-        return getCaseOperation.execute(jurisdictionId, caseTypeId, caseReference)
-            .flatMap(caseDetails ->
-                         verifyReadAccess(getCaseType(caseTypeId),
-                                          getUserRoles(),
-                                          caseDetails));
+        return this.execute(caseReference);
     }
 
     @Override
     public Optional<CaseDetails> execute(String caseReference) {
         return getCaseOperation.execute(caseReference)
             .flatMap(caseDetails ->
-                         verifyReadAccess(getCaseType(caseDetails.getCaseTypeId()),
-                                          getUserRoles(),
-                                          caseDetails));
+                verifyReadAccess(getCaseType(caseDetails.getCaseTypeId()),
+                    getUserRoles(caseDetails.getCaseTypeId()),
+                    caseDetails));
     }
 
     private CaseType getCaseType(String caseTypeId) {
         return caseDefinitionRepository.getCaseType(caseTypeId);
     }
 
-    private Set<String> getUserRoles() {
-        return userRepository.getUserRoles();
+
+    private Set<String> getUserRoles(String caseTypeId) {
+        return Sets.union(userRepository.getUserRoles(), caseRoleRepository.getCaseRoles(caseTypeId));
```

Comment:
I believe this is wrong, the case roles defined as part of the case type are only the list of valid case roles, but they do not represent the case role owned by the current user. Instead, the user case role (if any) must be read from the table `case_users` for the current case ID and user ID.